### PR TITLE
FOGL-9019: addition of mandatory and minimum for parameters

### PR DESCRIPTION
**Pull request by pintomax**
FOGL-9019: addition of mandatory and minimum for parameters

### CODE REVIEW

File: plugin.cpp [hunk 1]
```diff
@@ -46,35 +46,43 @@ static const char *default_config = QUOTE({
...
 		"displayName" : "Asset Name prefix",
-		"order" : "1"
+		"order" : "1",
+		"mandatory": "true"
 		},
 	"master_id" : {
 		"description" : "Master Link Id",
 		"type" : "integer",
 		"default" : DEFAULT_MASTER_LINK_ID,
 		"displayName" : "Master link Id",
```
Comment (praveen-garg) [inline plugin.cpp:56]: Id -> ID

File: plugin.cpp [hunk 2]
```diff
@@ -46,35 +46,43 @@ static const char *default_config = QUOTE({
...
-		"order" : "4"
+		"order" : "4",
+		"maximum" : "65000",
+		"minimum" : "1"
 		},
 	"outstation_id" : {
 		"description" : "Outstation Link Id",
 		"type" : "integer",
 		"default" : DEFAULT_OUTSTATION_ID,
 		"displayName" : "Outstation link Id",
```
Comment (praveen-garg) [inline plugin.cpp:82]: Id -> ID

File: plugin.cpp [hunk 3]
```diff
@@ -46,35 +46,43 @@ static const char *default_config = QUOTE({
 		"type" : "string",
 		"default" : DEFAULT_ASSETNAME_PREFIX,
 		"displayName" : "Asset Name prefix",
-		"order" : "1"
+		"order" : "1",
+		"mandatory": "true"
 		},
 	"master_id" : {
 		"description" : "Master Link Id",
```
Comment (praveen-garg) [inline plugin.cpp:53]: Id -> ID

File: plugin.cpp [hunk 4]
```diff
@@ -46,35 +46,43 @@ static const char *default_config = QUOTE({
...
 		"type" : "integer",
 		"default" : DEFAULT_TCP_PORT,
 		"displayName" : "Outstation port",
-		"order" : "4"
+		"order" : "4",
+		"maximum" : "65000",
+		"minimum" : "1"
 		},
 	"outstation_id" : {
 		"description" : "Outstation Link Id",
```
Comment (praveen-garg) [inline plugin.cpp:79]: Id -> ID

File: plugin.cpp [hunk 5]
```diff
@@ -46,35 +46,43 @@ static const char *default_config = QUOTE({
...
 		"type" : "integer",
 		"default" : DEFAULT_OUTSTATION_ID,
 		"displayName" : "Outstation link Id",
-		"order" : "5"
+		"order" : "5",
+		"maximum" : "65519",
+		"minimum" : "1"
```
[the remark this answers is not in the document]
Comment (pintomax) [reply plugin.cpp:85]: the indentifiers is > 0

File: plugin.cpp [hunk 6]
```diff
@@ -46,35 +46,43 @@ static const char *default_config = QUOTE({
...
 		},
 	"outstation_tcp_port" : {
 		"description" : "Outstation TCP/IP port",
 		"type" : "integer",
 		"default" : DEFAULT_TCP_PORT,
 		"displayName" : "Outstation port",
-		"order" : "4"
+		"order" : "4",
+		"maximum" : "65000",
+		"minimum" : "1"
```
[the remark this answers is not in the document]
Comment (pintomax) [reply plugin.cpp:76]: the indentifiers is > 0

File: plugin.cpp [hunk 7]
```diff
@@ -46,35 +46,43 @@ static const char *default_config = QUOTE({
...
+		"mandatory": "true"
 		},
 	"outstation_tcp_port" : {
 		"description" : "Outstation TCP/IP port",
 		"type" : "integer",
 		"default" : DEFAULT_TCP_PORT,
 		"displayName" : "Outstation port",
-		"order" : "4"
+		"order" : "4",
+		"maximum" : "65000",
```
[the remark this answers is not in the document]
Comment (pintomax) [reply plugin.cpp:75]: Simulator uses the limits added here and above

**Review by praveen-garg — APPROVED** [review]

**Pull request merged**